docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,232 @@
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log
+  Plugin->>Mapper: File transfer [HTTP]
+  Plugin->>Mapper: Status: uploading
+  alt if necessary
+    Mapper->>Cloud: Status: uploading
+  end
+  Mapper->>Cloud: Upload log file [HTTP]
+  Cloud-->> Mapper: Return the http endpoint of the uploaded file [HTTP]
+  alt if necessary
+    Mapper->>Cloud: Status: successful
+    Mapper->>Cloud: Cloud-specific something if any
+  end
+```
+
+### Failed flows
+
+a. Failure on the plugin side
+
+The status transition: `init` -> `executing` -> `failed`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log (FAILED!)
+  Plugin->>Mapper: Status: failed with reason
+  alt if necessary
+    Mapper->>Cloud: Status: failed with reason
+  end
+```
+
+b. Failure on uploading log (mapper side)
+
+The status transition: `init` -> `executing` -> `uploading` -> `failed`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log
+  Plugin->>Mapper: File transfer [HTTP]
+  Plugin->>Mapper: Status: uploading
+  alt if necessary
+    Mapper->>Cloud: Status: uploading
+  end
+  Mapper->>Cloud: Upload log file [HTTP] (FAILED!)
+  Cloud-->> Mapper: Return the http endpoint of the uploaded file [HTTP] (FAILED CODE!)
+  alt if necessary
+    Mapper->>Cloud: Cloud-specific Status failed
+  end
+```
+
+## Usage
+
+TBD
+
+## Logging
+
+The `tedge-log-plugin` reports progress and errors to the OS journal which can be retrieved using `journalctl`.
+
+
+## Not clear / undecided
+
+* Renaming the toml file from `c8y-log-plugin.toml` to `tedge-log-plugin.toml`.
+* Supported operation. Create automatically `c8y_LogfileRequest` as it is now? Or ask user to create it?
+* The location of the toml file. Change from `/etc/tedge/c8y/c8y-log-plugin.toml` to `/etc/tedge/plugins/tedge-log-plugin.toml`?
+
+## Check points (MQTT topic names and payloads)
+
+* Supported log type declaration format. CSV vs JSON? I prefer JSON as it's easy to extend fields in the future.
+* Type declaration topic: `te/device/main///cmd/log_upload`?
+* For receiving operation from mapper: `te/device/main///cmd/log_upload`?
```

Comment:
The command topic should have the command id as the last field (including on the initial creation of the command).

```suggestion
* For receiving operation from mapper: `te/device/main///cmd/log_upload/<cmd_id>`?
```

The command topic without the `/<cmd_id>` suffix, e.g. `te/device/main///cmd/log_upload`, is used by a component to signal that the `log_upload` capability is supported on the main device.